### PR TITLE
Add option to show recommeded fee on checkout invoice

### PR DESCRIPTION
**Pull request by bolatovumar**
Address #1036

### CODE REVIEW

File: BTCPayServer/Controllers/InvoiceController.UI.cs [hunk 1]
```diff
@@ -224,7 +224,11 @@ private async Task<PaymentModel> GetInvoiceModel(string invoiceId, PaymentMethod
                 ? (accounting.Due.ToDecimal(MoneyUnit.BTC) *
                    (1m + (changelly.AmountMarkupPercentage / 100m)))
                 : (decimal?)null;
-
+            var recommendedFeeText = storeBlob.RecommendedFeeText != null 
+                ? storeBlob.RecommendedFeeText
+                    .Replace("{amount}", paymentMethodDetails.GetFeeRate().ToString(CultureInfo.InvariantCulture), StringComparison.OrdinalIgnoreCase)
+                    .Replace("{crypto_unit_name}", network.CryptoCode == "LTC" ? "litoshi" : "satoshi", StringComparison.OrdinalIgnoreCase)
```
Comment (Kukks) [inline BTCPayServer/Controllers/InvoiceController.UI.cs]: No specific coin logic here. Add a property in `BTCPayNetworkBase` then do `network.BaseUnitName`.

Comment (Kukks) [reply BTCPayServer/Controllers/InvoiceController.UI.cs]: You should also check if the fee rate returns 0 and hide message if so(lightning for example)

Comment (bolatovumar) [reply BTCPayServer/Controllers/InvoiceController.UI.cs]: @Kukks message is hidden for lightning already. It's only shown for BTC/LTC.

Comment (bolatovumar) [reply BTCPayServer/Controllers/InvoiceController.UI.cs]: > No specific coin logic here. Add a property in `BTCPayNetworkBase` then do `network.BaseUnitName`.

Done.

File: BTCPayServer.Common/Altcoins/BTCPayNetworkProvider.Dogecoin.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@ public void InitDogecoin()
             {
                 CryptoCode = nbxplorerNetwork.CryptoCode,
                 DisplayName = "Dogecoin",
+                BaseUnitName = "dogetoshi",
```
Comment (NicolasDorier) [inline BTCPayServer.Common/Altcoins/BTCPayNetworkProvider.Dogecoin.cs]: lol did not know this one. @astupidmoose would approve.

**Comment by NicolasDorier** [comment]
I propose simplifying: 

1. Just use `sat/byte` as this is show in every wallet in all languages.
2. The sentence should be localized like other strings in checkout, so people can just translate it on their language via Transifex.
3. Should be activated by default for new stores.

File: BTCPayServer/Views/Invoice/Checkout.cshtml [hunk 3]
```diff
@@ -185,6 +185,9 @@
             showEmailForm: function(){
                 return this.srvModel.requiresRefundEmail && (!this.srvModel.customerEmail || !this.validateEmail(this.srvModel.customerEmail)) && !this.invoiceUnpayable && !this.invoicePaid;
             },
+            showRecommendedFee: function(){
+                return (this.srvModel.cryptoCode === 'BTC' || this.srvModel.cryptoCode === 'LTC') && this && this.srvModel.showRecommendedFee && !!this.srvModel.recommendedFeeText && !this.srvModel.isLightning;
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Invoice/Checkout.cshtml]: I think you should remove `this.srvModel.cryptoCode === 'BTC' || this.srvModel.cryptoCode === 'LTC'` 

Maybe test if the fee rate is 0, if so, then don't show it.

Comment (bolatovumar) [reply BTCPayServer/Views/Invoice/Checkout.cshtml]: makes sense

**Comment by bolatovumar** [comment]
> I propose simplifying:
> 
> 1. Just use `sat/byte` as this is show in every wallet in all languages.
> 2. The sentence should be localized like other strings in checkout, so people can just translate it on their language via Transifex.
> 3. Should be activated by default for new stores.
> 
> It means that shitcoin can't customize it, but this is fine, we should not care much about shitcoins.

Just to clarify point 2:
Do you mean I should remove the fee suggestion sentence from the checkout experience options and just have the string hardcoded in localization JSON files? 

And point 1:
Remove the satoshi/litoshi distinction and have it shown as "sat/byte" instead for all coins?

File: BTCPayServer/Views/Invoice/Checkout.cshtml (hunk 3 above)
Comment (bolatovumar) [reply BTCPayServer/Views/Invoice/Checkout.cshtml]: @NicolasDorier checking if the fee rate is 0 would require me to add the fee rate to the model. That's OK by you then?

Comment (NicolasDorier) [reply BTCPayServer/Views/Invoice/Checkout.cshtml]: yeah

**Comment by NicolasDorier** [comment]
> Do you mean I should remove the fee suggestion sentence from the checkout experience options and just have the string hardcoded in localization JSON files?

Yes I think it is better, so we can easily push the translation to people on transifex.

> Remove the satoshi/litoshi distinction and have it shown as "sat/byte" instead for all coins?

I think we should, because I don't think we can do both, localization and being shitcoin specific at same time.

File: BTCPayServer.Common/Altcoins/BTCPayNetworkProvider.Dogecoin.cs (hunk 2 above)
Comment (bolatovumar) [reply BTCPayServer.Common/Altcoins/BTCPayNetworkProvider.Dogecoin.cs]: I made it up. Was tempted to call it "doge" since 1 Doge is always 1 Doge.

**Comment by bolatovumar** [comment]
Looks like the build is failing due to some CircleCI issues again. Not related to my commit.

**Comment by Eskyee** [comment]
builds fails from here https://circleci.com/gh/btcpayserver/btcpayserver/3258

**Comment by bolatovumar** [comment]
@Eskyee that one is for pull request #1077. Mine is #1076.

**Comment by Eskyee** [comment]
thanks, bolatovumar,  ;)

**Comment by bolatovumar** [comment]
Build fails due to this error. Anything can be done to trigger a build again?

Screen Shot 2019-10-13 at 9 58 29 AM

**Comment by Eskyee** [comment]
Build fails, I also get the same as above... @bolatovumar

**Comment by NicolasDorier** [comment]
@bolatovumar can you rebase this PR? this would fix the build issue.

**Comment by bolatovumar** [comment]
Resolved the conflict.

**Comment by bolatovumar** [comment]
@NicolasDorier anything else needed to be done here before this can be merged-in?

**Comment by NicolasDorier** [comment]
@bolatovumar perfect will review today. I want this in the next version.

**Comment by NicolasDorier** [comment]
This PR is simpler than I expected, I completely forgot I was fetching the fees when creating the invoice. This is awesome.

File: BTCPayServer/wwwroot/locales/en.json [hunk 4]
```diff
@@ -47,5 +47,6 @@
   "Pay with CoinSwitch": "Pay with CoinSwitch",
   "Pay with Changelly": "Pay with Changelly",
   "Close": "Close",
-  "NotPaid_ExtraTransaction": "The invoice hasn't been paid in full. Please send another transaction to cover amount Due."
+  "NotPaid_ExtraTransaction": "The invoice hasn't been paid in full. Please send another transaction to cover amount Due.",
+  "Recommended_Fee": "Recommended fee: {{feeRate}} sat/byte"
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/locales/en.json]: @britttttk when there is a new sentence, we need to do something at transifex level?

Comment (bolatovumar) [reply BTCPayServer/wwwroot/locales/en.json]: @NicolasDorier yeah, I assumed the wording of this sentence can be changed at a later date. Not sure how it actually would be done though.

**Pull request merged**